pubsub/rocketmq/metadata.go

```diff
@@ -14,68 +14,166 @@ limitations under the License.
 package rocketmq
 
 import (
-	"errors"
 	"fmt"
+	"strconv"
 
 	"github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/components-contrib/pubsub"
-	"github.com/dapr/kit/logger"
-)
-
-var (
-	ErrRocketmqPublishMsg         = errors.New("rocketmq publish msg error")
-	ErrRocketmqValidPublishMsgTyp = errors.New("rocketmq publish msg error, invalid msg type")
 )
 
 const (
 	metadataRocketmqTag           = "rocketmq-tag"
 	metadataRocketmqKey           = "rocketmq-key"
 	metadataRocketmqShardingKey   = "rocketmq-shardingkey"
+	metadataRocketmqQueue         = "rocketmq-queue"
 	metadataRocketmqConsumerGroup = "rocketmq-consumerGroup"
 	metadataRocketmqType          = "rocketmq-sub-type"
 	metadataRocketmqExpression    = "rocketmq-sub-expression"
 	metadataRocketmqBrokerName    = "rocketmq-broker-name"
+	metadataRocketmqQueueID       = "rocketmq-queue-id"
+)
+
+type QueueSelectorType string
+
+const (
+	HashQueueSelector       QueueSelectorType = "hash"
+	RandomQueueSelector     QueueSelectorType = "random"
+	ManualQueueSelector     QueueSelectorType = "manual"
+	RoundRobinQueueSelector QueueSelectorType = "roundRobin"
+	DaprQueueSelector       QueueSelectorType = "dapr"
 )
 
+// RocketMQ Go Client Options
 type rocketMQMetaData struct {
-	// rocketmq Credentials
-	AccessKey  string `mapstructure:"accessKey"`
-	SecretKey  string `mapstructure:"secretKey"`
-	NameServer string `mapstructure:"nameServer"`
-	// Deprecated: use ProducerGroup instead.
-	GroupName     string `mapstructure:"groupName"`
-	ProducerGroup string `mapstructure:"producerGroup"`
-	NameSpace     string `mapstructure:"nameSpace"`
-	// consumer group rocketmq's subscribers
-	ConsumerGroup     string `mapstructure:"consumerGroup"`
-	ConsumerBatchSize int    `mapstructure:"consumerBatchSize"`
+	// rocketmq instance name, it will be registered to the broker
+	InstanceName string `mapstructure:"instanceName"`
+	// consumer group name
+	ConsumerGroupName string `mapstructure:"consumerGroupName"`
+	// producer group name
+	ProducerGroupName string `mapstructure:"producerGroupName"`
+	// rocketmq namespace
+	NameSpace string `mapstructure:"nameSpace"`
 	// rocketmq's name server domain
 	NameServerDomain string `mapstructure:"nameServerDomain"`
-	// msg's content-type
-	ContentType string `mapstructure:"content-type"`
-	// retry times to connect rocketmq's broker
+	// rocketmq's name server
+	NameServer string `mapstructure:"nameServer"`
+	// rocketmq Credentials
+	AccessKey     string `mapstructure:"accessKey"`
+	SecretKey     string `mapstructure:"secretKey"`
+	SecurityToken string `mapstructure:"securityToken"`
+	// retry times to send msg to broker
 	Retries int `mapstructure:"retries"`
-	// Deprecated: send msg timeout to connect rocketmq's broker, nanoseconds
-	SendTimeOut int `mapstructure:"sendTimeOut"`
-	// send msg timeout to connect rocketmq's broker, seconds
-	SendTimeOutSec int `mapstructure:"sendTimeOutSec"`
```

Comment:
This has been removed too?

Reply:
Per below, "sendTimeOut" should have been removed (deprecated), while "sendTimeOutSec" needs to stay.

Reply:
ok，I'll drop the sendTimeOut renaming.